lnbits/wallets/blink.py

```diff
@@ -0,0 +1,481 @@
+                .get("me", {})
+                .get("defaultAccount", {})
+                .get("wallets", [])
+            )
+            btc_wallet_ids = [
+                wallet["id"] for wallet in wallets if wallet["walletCurrency"] == "BTC"
+            ]
+
+            if not btc_wallet_ids:
+                raise ValueError("BTC Wallet not found")
+
+            self._wallet_id = btc_wallet_ids[0]
+            return self._wallet_id
+        except Exception as exc:
+            logger.warning(exc)
+            raise ValueError(f"Unable to connect to '{self.endpoint}'") from exc
+
+
+class BlinkGrafqlQueries(BaseModel):
+    balance_query: str
+    invoice_query: str
+    payment_query: str
+    status_query: str
+    wallet_query: str
+    tx_query: str
+    my_updates_query: str
+
+
+q = BlinkGrafqlQueries(
+    balance_query="""
+        query Me {
+          me {
+            defaultAccount {
+              wallets {
+                walletCurrency
+                balance
+              }
+            }
+          }
+        }
+        """,
+    invoice_query="""
+        mutation LnInvoiceCreateOnBehalfOfRecipient(
+          $input: LnInvoiceCreateOnBehalfOfRecipientInput!
+        ) {
+          lnInvoiceCreateOnBehalfOfRecipient(input: $input) {
+            invoice {
+              paymentRequest
+              paymentHash
+              paymentSecret
+              satoshis
+            }
+            errors {
+              message
+            }
+          }
+        }
+        """,
+    payment_query="""
+        mutation LnInvoicePaymentSend($input: LnInvoicePaymentInput!) {
+          lnInvoicePaymentSend(input: $input) {
+            status
+            errors {
+              message
+              path
+              code
+            }
+          }
+        }
+        """,
+    status_query="""
+        query InvoiceByPaymentHash($walletId: WalletId!, $paymentHash: PaymentHash!) {
+          me {
+            defaultAccount {
+              walletById(walletId: $walletId) {
+                invoiceByPaymentHash(paymentHash: $paymentHash) {
+                  ... on LnInvoice {
+                    paymentStatus
+                  }
+                }
+              }
+            }
+          }
+        }
+        """,
+    wallet_query="""
+        query me {
+          me {
+            defaultAccount {
+              wallets {
+                id
+                walletCurrency
+              }
+            }
+          }
+        }
+        """,
+    tx_query="""
+        query TransactionsByPaymentHash(
+          $walletId: WalletId!
+          $transactionsByPaymentHash: PaymentHash!
+        ) {
+          me {
+            defaultAccount {
+              walletById(walletId: $walletId) {
+                walletCurrency
+                ... on BTCWallet {
+                  transactionsByPaymentHash(paymentHash: $transactionsByPaymentHash) {
+                    settlementFee
+                    status
```

Comment:
@motorina0  adding `direction` here will help to clarify the fee refund case when there are two entries for a transaction hash.
Only the `settlementFee` of the `SEND` direction should be counted. The refund is marked as `RECEIVE`. The actual history with the refund is deiplayed clearley in the Blink Wallet connected to the account so can double check there.

Reply:
Done